src/Client/src/Asp.Versioning.Http.Client/System.Net.Http/HttpResponseMessageExtensions.cs

```diff
@@ -69,6 +70,67 @@ public static SunsetPolicy ReadSunsetPolicy( this HttpResponseMessage response )
         return policy;
     }
 
+    /// <summary>
+    /// Gets an API deprecation policy from the HTTP response.
+    /// </summary>
+    /// <param name="response">The <see cref="HttpResponseMessage">HTTP response</see> to read from.</param>
+    /// <returns>A new <see cref="DeprecationPolicy">deprecation policy</see>.</returns>
+    public static DeprecationPolicy ReadDeprecationPolicy( this HttpResponseMessage response )
+    {
+        ArgumentNullException.ThrowIfNull( response );
+
+        var headers = response.Headers;
+        var date = default( DateTimeOffset );
+        DeprecationPolicy policy;
+
+        if ( headers.TryGetValues( Deprecation, out var values ) )
+        {
+            var culture = CultureInfo.CurrentCulture;
+
+            foreach ( var value in values )
+            {
+                var split = value.Trim( '@' );
```

Comment:
We don't want to use `String.Trim` here because it will allocate. We can do the test by just checking the first character. Also, as strange as it might sound, a header _can_ have an empty value.

The following should work for all platforms without allocating:

```c#
if ( value.Length < 2 || value[0] != '@' )
{
    continue;
}
```